sqlx-core/src/postgres/listener.rs

```diff
@@ -125,6 +124,7 @@ impl PgListener {
     /// The channel name is quoted here to ensure case sensitivity.
     pub async fn unlisten(&mut self, channel: &str) -> Result<(), Error> {
         self.connection()
+            .await?
```

Comment:
This and `unlisten_all()` don't technically need to reconnect right away; if the connection is dead, they just need to delete the channels from the local set.

In fact, this is rather wasteful as `.connection()` will internally reconnect and then execute a `LISTEN` query for every channel and then immediately execute `UNLISTEN {channel}`.

Reply:
done